packages/pigeon/CONTRIBUTING.md

```diff
@@ -0,0 +1,74 @@
+# Pigeon Contributor's Guide
+
+## Description
+
+Pigeon is a code generation tool that adds type safety to Flutter’s Platform
+Channels.  This document serves as an overview of how it functions to help
+people who would like to contribute to the project.
+
+## State Diagram
+
+Pigeon generates a temporary file in its LaunchIsolate, then launches another
+isolate that uses dart:mirrors to parse the generated file, creating an AST,
+then running code generators with that AST.
+
+![State Diagram](./docs/pigeon_state.png)
+
+## Generated Source Code Example
+
+This is what the generated code looks like:
```

Comment:
Not sure I understood this section

Reply:
I expanded on the description sentence and moved it down lower in the document since it's not as important.